tools/ml-audio-content-moderation/README.md

```diff
@@ -0,0 +1,338 @@
+````
+
+3. Set your terminal to use that project
+
+````
+gcloud config set project $PROJECT
+````
+
+Deployment
+==========
+
+Step 1: [Deploy the App Engine frontend](#step-1)
+
+Step 2a: [Use gcloud for remaining resources](#step-2a)
+
+Step 2b: [Use terraform for remaining resources](#step2-b)
+
+Step 3: [View Results](#view-results)
+
+### Step 1
+<h3>Clone the repository and change your directory into the root folder</h3>
+
+1. In your terminal type the following to clone the professional-services repository: 
+
+````
+git clone https://github.com/GoogleCloudPlatform/professional-services
+````
+
+2. Change directories into this project
+
+
+`````
+cd tools/ml-audio-content-moderation/
+`````
+
+<h3>Create Project Resources</h3>
+1. Change directories into the frontend App Engine application folder.
+
+````
+cd app
+````
+
+2a. OPTIONAL: Compile the angular frontend application. You only need to do this if you altered
+the frontend in any way. Note that this requires [installing Angular](https://angular.io/cli) on 
+your device.
+
+````
+npm install -g @angular/cli
+````
+
+````
+cd angular/
+````
+
+````
+ng build --prod
+````
+
+````
+cd ..
+````
+
+2b. There is also an [Open Sourced moderator UI](https://github.com/conversationai/conversationai-moderator)
+which contains detailed features for sorting through results from the Perspective API. Note that
+this will not display any results from the NLP API or a transcript itself, but you may add in these
+additional features if you wish. This is an alternative to the frontend in the `app` folder in this
+repository.
+
+
+3. Deploy the application.
+
+````
+gcloud app deploy
+````
+
+You will be prompted to use a region to serve the location from. You may pick any region, but you 
+cannot change this value later.
+
+You can verify that the app was deployed correctly by navigating to 
+https://`$PROJECT`.appspot.google.com. You should see the following UI:
+
+![text](img/app_home.png?raw=true)
+
+
+### Step 2a
+<h3>Enable APIs for your Project</h3>
+1. Enable APIs
+
+````
+gcloud services enable language.googleapis.com
+gcloud services enable speech.googleapis.com
+gcloud services enable cloudfunctions.googleapis.com
+gcloud services enable commentanalyzer.googleapis.com
+gcloud services enable cloudscheduler.googleapis.com
+````
+
+
+2. Change directories to be at the root directory again.
+
+````
+cd ..
+````
+
+<h3>Create PubSub resources</h3>
+
+1. Create PubSub topic named stt_queue
+
+````
+export TOPIC_NAME=stt_queue
+
+gcloud pubsub topics create $TOPIC_NAME
+````
+
+2. Create subscription to stt_queue topic
+
+````
+export SUBSCRIPTION_NAME=pull_stt_ids
+
+gcloud pubsub subscriptions create $SUBSCRIPTION_NAME --topic=$TOPIC_NAME
+````
+
+3. Generate a static UUID that you will need in each of your bucket names to ensure that they are 
+unique.
+
+3a. First install uuidgen. If it is already installed, skip this step.
+
+````
+sudo apt-get install uuid-runtime
```

Comment:
Agreed, it's probably installed by default. 

I think since they're probably doing the whole setup in their own terminal, they probably don't need to open up the shell for just generating a unique ID. I think it's better to just tell them it needs to be a unique ID, which they'll find out immediately if it fails when they do the gsutil commands. WDYT?

Reply:
@jaketf